ada-project-docs/wave_04.md

```diff
@@ -0,0 +1,47 @@
+# OPTIONAL Wave 4: Enhancements
+
+These really are **optional** - if you've gotten here and you have time left, that means you're moving speedy fast!
+
+## Enhancements
+
+### More Query Parameters
+The following 3 _optional_ query parameters:
+
+| Name   | Value   | Description
+|--------|---------|------------
+| `sort` | string  | Sort objects by this field, in ascending order
+| `n`    | integer | Number of responses to return per page
+| `p`    | integer | Page of responses to return
+
+should additionally be accepted by the following three endpoints:
+- `GET /video`
+- `GET /customers/<id>/history`
+- `GET /videos/<id>/history`
+
+So, for an API endpoint like `GET /customers`, the following requests should be valid:
+- `GET /customers`: All customers, sorted by ID
+- `GET /customers?sort=name`: All customers, sorted by name
+- `GET /customers?n=10&p=2`: Customers 11-20, sorted by ID
+- `GET /customers?sort=name&n=10&p=2`: Customers 11-20, sorted by name
```

Comment:
It looks like the examples are the same from Wave 3, I think it could be clearer for students if these are updated to use one of the endpoints listed above.